### ISSUE TITLE
[Epic] Add Gaudi support to InstructLab CLI, eval, and training

### ISSUE THREAD

**Issue by ktam3** [instructlab/instructlab#2218]
**Feature Overview (aka. Goal Summary)**
Implement Intel Gaudi support in InstructLab project, so Gaudi 2 and Gaudi 3 can be used for SDG, evaluation, and training.

**Goals (aka. expected user outcomes)**
InstructLab implements necessary features to run the RHEL AI pipeline on Torch device "hpu" (Habana Processing Unit aka Intel Gaudi)

**Acceptance Criteria**

- [x] #2219 - @tiran may have a PR already opened for this. Waiting on confirmation
- [ ] #2220 (11/13 Update: On Hold - assess if there is a need for this. Focus on Gaudi 3 profiles)
- [ ] #2221
- [x] #2222 - This is no longer relevant, issue has been closed 
- [x] https://github.com/instructlab/training/issues/239

**Comment by JamesKunstle** [comment]
Started working on this a few weeks ago (pre Sept. 20th). Ran into a problem porting our FSDP code to work on Gaudi 2 machine. At the time, I was working w/ intel engineers to solve- need to revisit this.

As of now, we've got access to Gaudi 2 machines via IBM, and will continue testing. 

For the training repo, this is tracked in training#239

**Comment by ktam3** [comment]
Closing this issue. The other remaining issues are tracked in another epic
